### PR TITLE
feat(schema): added schema for templates

### PR DESCRIPTION
**Pull request by LeCuay**
# Summary

Since this project relies on people editing JSON directly I thought it'd be a good idea to have an schema for it.

I added it to the `.vscode` folder and tweaked the settings. Not sure if this is the way but since it was there already it felt natural.

The conditionals work by automatically detecting type and then looking for required keys.

> It's not intended for this schema to replace documentation. It just to force checks when typing and improve intellisense.

### CODE REVIEW

**Comment by LeCuay** [comment]
Just realized it's missing `else` optional property for `match` but what do you think so far?

**Comment by EddieDover** [comment]
> Just realized it's missing `else` optional property for `match` but what do you think so far?

Love it so far. I've been working on a visual creator but as you can imagine it's quite the chore and finding time to peck away at it between other responsibilities is rough.  This PR goes a long way to helping people make sane templates until it's finished.

**Comment by EddieDover** [comment]
Any issues holding off on this until we get the others PRs finalized? That way you can add the rowspan and other updates?

**Comment by LeCuay** [comment]
> Any issues holding off on this until we get the others PRs finalized? That way you can add the rowspan and other updates?

No problem at all! Waiting so we have all implemented sounds good

**Comment by LeCuay** [comment]
Added else statement

**Comment by LeCuay** [comment]
Added all new properties as per #39 and #30.

File: .vscode/template.schema.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+              "type": "boolean"
+            },
+            "rowspan": {
+              "description": "Controls the row span of the cells in the column. You need also the corresponding empty rows and type 'span'.",
+              "type": "number"
+            },
+            "colspan": {
+              "description": "Controls the column span of the cells in the column.",
+              "type": "number"
+            },
```
Comment (EddieDover) [inline .vscode/template.schema.json:103]:
```suggestion
```

Let's drop this until colspan is re-implemented

**Review by EddieDover — CHANGES_REQUESTED** [review]
Knock out the colspan reference and this is good to go.

**Comment by LeCuay** [comment]
> Knock out the colspan reference and this is good to go.

Donezo

**Review by EddieDover — APPROVED** [review]

**Pull request merged**